### ISSUE TITLE
`MockOsSysCalls::setsockopt` has unwarranted ASSERT

### ISSUE THREAD

**Issue by efimki** [envoyproxy/envoy#19079]
*Title*: `MockOsSysCalls::setsockopt` has unwarranted ASSERT

*Description*:
`MockOsSysCalls::setsockopt` has `ASSERT(optlen == sizeof(int));` but `setsockopt` API allows socket options to have different length. This breaks our internal test that sets 16 byte socket option.

*Repro steps*:

The following test triggers an ASSERT in debug build:

```
TEST(MockOsSysCallsTest, TestSetSockopt) {
  Envoy::Api::MockOsSysCalls os_sys_calls;
  char big_sock_opt[11] = "bigsockopt";
  os_sys_calls.setsockopt(1, 2, 3, big_sock_opt, sizeof(big_sock_opt));
}
```

**Comment by efimki** [comment]
I'm not a `setsockopt` expert. 
Maybe there should be an assert that `optval != nullptr`, and/or `optlen != 0`.

**Comment by RyanTheOptimist** [comment]
cc: @yanavlasov

**Comment by RyanTheOptimist** [comment]
I can't quite find an appropriate label for this. @yanavlasov, any suggestions?

**Comment by ggreenway** [comment]
@efimki feel free to create a PR to fix

**Comment by efimki** [comment]
Created https://github.com/envoyproxy/envoy/pull/19126

**Comment by efimki** [comment]
Fixed by #19126
